### PR TITLE
BigQuery: Add ability to pass in a table ID instead of a query to the %%bigquery magic.

### PR DESCRIPTION
**Pull request by shubha-rajan**
Third of 3 PRs towards resolving #9105 as described in review for #9147
Screenshot of feature in notebook:
image

### CODE REVIEW

File: bigquery/google/cloud/bigquery/magics.py [hunk 1]
```diff
@@ -434,6 +444,26 @@ def _cell_magic(line, query):
...
+    error = None
+
+    if not re.search(r"\s", query.rstrip()):
+        table_id = query.rstrip()
+
+        try:
+            rows = client.list_rows(table_id, max_results=max_results)
+        except Exception as ex:
+            error = str(ex)
+        if error:
```
Comment (crwilcox) [inline bigquery/google/cloud/bigquery/magics.py]: could this be simplified to return from the exception?

File: bigquery/google/cloud/bigquery/magics.py [hunk 2]
```diff
@@ -434,6 +444,26 @@ def _cell_magic(line, query):
     else:
         max_results = None
 
+    error = None
+
+    if not re.search(r"\s", query.rstrip()):
```
Comment (crwilcox) [inline bigquery/google/cloud/bigquery/magics.py]: What scenario is this protecting against? Is there a scenario where unicode strings will not remove space characters via rstrip?

What do we expect in the else case?

Comment (crwilcox) [reply bigquery/google/cloud/bigquery/magics.py]: Related: Does this repository enforce code coverage? Is there a case where we test there being unstoppable whitespace and not taking this branch?

Comment (shubha-rajan) [reply bigquery/google/cloud/bigquery/magics.py]: This condition is actually testing for queries that contain whitespace characters that aren't removed by rstrip (so any whitespace that isn't a trailing newline). The assumption being made (as described in #9105 ) is that anything containing whitespace is a SQL query and won't take this branch, while anything string without whitespace is a table_id and will take this branch. 

So anything regular SQL query would fall into the else case. There are tests that check whether query strings without spaces will be interpreted as table IDs and a test that checks if a string without whitespace that isn't a valid table_id raises an appropriate error message.

File: bigquery/google/cloud/bigquery/magics.py (hunk 1 above)
Comment (shubha-rajan) [reply bigquery/google/cloud/bigquery/magics.py]: it definitely can.  fixed in 7a13ca4

File: bigquery/google/cloud/bigquery/magics.py (hunk 2 above)
Comment (shubha-rajan) [reply bigquery/google/cloud/bigquery/magics.py]: I can add a comment in the code if that would be helpful.

File: bigquery/google/cloud/bigquery/magics.py [hunk 3]
```diff
@@ -434,6 +444,21 @@ def _cell_magic(line, query):
     else:
         max_results = None
 
+    if not re.search(r"\s", query.rstrip()):
+        table_id = query.rstrip()
```
Comment (plamut) [inline bigquery/google/cloud/bigquery/magics.py]: BTW,  stripping whitespace from the same value again is not necessary, we could store the stripped string into a variable the first time we do it.

Comment (shubha-rajan) [reply bigquery/google/cloud/bigquery/magics.py]: fixed in d663104

File: bigquery/google/cloud/bigquery/magics.py [hunk 4]
```diff
@@ -445,24 +470,14 @@ def _cell_magic(line, query):
         value = int(args.maximum_bytes_billed)
         job_config.maximum_bytes_billed = value
 
-    error = None
     try:
         query_job = _run_query(client, query, job_config=job_config)
     except Exception as ex:
-        error = str(ex)
+        return _print_error(str(ex), args.destination_var)
```
[the remark this answers is not in the document]
Comment (shubha-rajan) [reply bigquery/google/cloud/bigquery/magics.py]: fixed in d663104

File: bigquery/google/cloud/bigquery/magics.py (hunk 2 above)
Comment (shubha-rajan) [reply bigquery/google/cloud/bigquery/magics.py]: added comment in d663104

**Pull request merged**